plugins/inputs/whois/whois_test.go

```diff
@@ -0,0 +1,186 @@
+package whois
+
+import (
+	"errors"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/likexian/whois"
+	"github.com/likexian/whois-parser"
+	"github.com/stretchr/testify/require"
+)
+
+// Make sure Whois implements telegraf.Input
+var _ telegraf.Input = &Whois{}
+
+func ptr(t time.Time) *time.Time {
+	return &t
+}
+
+func TestSimplifyStatus(t *testing.T) {
+	tests := []struct {
+		input    []string
+		expected string
+	}{
+		{[]string{"clientTransferProhibited"}, "LOCKED"},
+		{[]string{"pendingDelete"}, "PENDING DELETE"},
+		{[]string{"redemptionPeriod"}, "EXPIRED"},
+		{[]string{"active"}, "ACTIVE"},
+		{[]string{"registered"}, "REGISTERED"},
+		{[]string{"unknownStatus"}, "UNKNOWN"},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.expected, func(t *testing.T) {
+			result := simplifyStatus(tt.input)
+			require.Equal(t, tt.expected, result)
+		})
+	}
+}
+
+func TestWhoisConfigInitialization(t *testing.T) {
+	tests := []struct {
+		name               string
+		domains            []string
+		server             string
+		IncludeNameServers bool
+		timeout            config.Duration
+		expectErr          bool
+	}{
+		{
+			name:      "Valid Configuration",
+			domains:   []string{"example.com", "google.com"},
+			timeout:   config.Duration(10 * time.Second),
+			expectErr: false,
+		},
+		{
+			name:      "No Domains Configured",
+			domains:   nil,
+			timeout:   config.Duration(5 * time.Second),
+			expectErr: true,
+		},
+		{
+			name:      "Invalid Timeout (Zero Value)",
+			domains:   []string{"example.com"},
+			timeout:   config.Duration(0),
+			expectErr: false, // Should still work, default timeout can be applied
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			plugin := &Whois{
+				Domains: test.domains,
+				Timeout: test.timeout,
+				Server:  test.server,
+				Log:     testutil.Logger{},
+			}
+
+			err := plugin.Init()
+
+			if test.expectErr {
+				require.Error(t, err, "Expected error but got none")
+				return
+			}
+
+			require.NoError(t, err, "Unexpected error during Init()")
+
+			err = plugin.Gather(&testutil.Accumulator{})
+			require.NoError(t, err, "Unexpected error during Gather()")
```

Comment:
No need to do a gather when testing the Init method and you are doing tests for Gather in other tests.